security/pkg/pki/signingapi/client.go

```diff
@@ -0,0 +1,185 @@
+			return nil, fmt.Errorf("append ca cert to pool failed: %v", rootCACert)
+		}
+		return grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{RootCAs: rootCAPool})), nil
+
+	case networking.ClientTLSSettings_MUTUAL:
+		var credentialOption *creds.Options
+
+		if c.TLSClientSetting.CaCertificates == "" {
+			signingLog.Infof("missing config for CaCertificates. Using system cert pools instead")
+			systemCertPool, err := x509.SystemCertPool()
+			if err != nil {
+				signingLog.Errorf("get system cert pool failed: %v", err)
+				return nil, fmt.Errorf("get system cert pool failed: %v", err)
+			}
+			trans := credentials.NewClientTLSFromCert(systemCertPool, c.TLSClientSetting.Sni)
+			return grpc.WithTransportCredentials(trans), nil
+		}
+
+		credentialOption = &creds.Options{
+			CertificateFile:   c.TLSClientSetting.ClientCertificate,
+			KeyFile:           c.TLSClientSetting.PrivateKey,
+			CACertificateFile: c.TLSClientSetting.CaCertificates,
+		}
+		watcher, err := creds.WatchFiles(ctx.Done(), credentialOption)
+		if err != nil {
+			signingLog.Errorf("create certificate files watcher failed: %v", err)
+			return nil, fmt.Errorf("create certificate files watcher failed: %v", err)
+		}
+		return grpc.WithTransportCredentials(creds.CreateForClient(c.TLSClientSetting.Sni, watcher)), nil
+	default:
+		signingLog.Errorf("do not support TLS Client Mode: %v", c.TLSClientSetting.Mode)
+		return nil, fmt.Errorf("do not support TLS Client Mode: %v", c.TLSClientSetting.Mode)
+	}
+}
+
+// CreateCertificate call CreateCertificate of Istio Certificate Signing API
+func (c *Client) CreateCertificate(ctx context.Context,
+	req *v1alpha1.IstioCertificateRequest) (*v1alpha1.IstioCertificateResponse, error) {
+	if !c.IsConnected() {
+		return nil, fmt.Errorf("istio certificate request client still not connected")
+	}
+	timeoutCtx, cancel := context.WithTimeout(ctx, c.RequestTimeout)
+	defer cancel()
+	signingLog.Infof("invoke CreateCertificate to CA Address: %v", c.Address)
+	resp, err := c.ClientV1.CreateCertificate(timeoutCtx, req)
+	if err != nil {
+		signingLog.Errorf("invoke CreateCertificate failed: %v", err)
+		return nil, fmt.Errorf("invoke CreateCertificate failed: %v", err)
+	}
+
+	certChain := resp.GetCertChain()
+	if len(certChain) < 2 {
+		signingLog.Errorf("invalid CreateCertificate response: %v", resp.GetCertChain())
+		return nil, fmt.Errorf("invalid CreateCertificate response: %v", resp.GetCertChain())
+	}
+
+	return resp, nil
+}
+
+// IsConnected checking client is connected
+func (c *Client) IsConnected() bool {
+	return c.Conn != nil && c.ClientV1 != nil
+}
+
+func trackingTimeResposeInterceptor(ctx context.Context, method string, req, reply interface{},
+	cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+	start := time.Now()
+	err := invoker(ctx, method, req, reply, cc, opts...)
+	signingLog.Infof("Invoked RPC method=%s; Duration=%s; Error=%v", method, time.Since(start), err)
```

Comment:
nit: This seems like it may be too spammy log

Reply:
I'll set log level to DEBUG